pkg/skaffold/cluster/minikube_test.go

```diff
@@ -56,41 +58,60 @@ func TestClientImpl_IsMinikube(t *testing.T) {
 			expected:    true,
 		},
+		{
+			description:             "minikube without user flag",
+			kubeContext:             "test-cluster",
+			minikubeProfileCmd:      testutil.CmdRunOut("minikube profile list -o json --user=skaffold", fmt.Sprintf(profileStr, "test-cluster", "docker", "172.17.0.3", 8443)),
+			certPath:                filepath.Join(home, "foo", "ca.crt"),
+			expected:                false,
+			minikuneWithoutUserFalg: true,
+		},
 		{
 			description:        "cluster cert outside minikube dir",
 			kubeContext:        "test-cluster",
-			minikubeProfileCmd: testutil.CmdRunOut("minikube profile list -o json", fmt.Sprintf(profileStr, "test-cluster", "docker", "172.17.0.3", 8443)),
+			minikubeProfileCmd: testutil.CmdRunOut("minikube profile list -o json --user=skaffold", fmt.Sprintf(profileStr, "test-cluster", "docker", "172.17.0.3", 8443)),
 			certPath:           filepath.Join(home, "foo", "ca.crt"),
 			expected:           false,
 		},
 		{
 			description:        "minikube node ip matches api server url",
 			kubeContext:        "test-cluster",
 			serverURL:          "https://192.168.64.10:8443",
-			minikubeProfileCmd: testutil.CmdRunOut("minikube profile list -o json", fmt.Sprintf(profileStr, "test-cluster", "hyperkit", "192.168.64.10", 8443)),
+			minikubeProfileCmd: testutil.CmdRunOut("minikube profile list -o json --user=skaffold", fmt.Sprintf(profileStr, "test-cluster", "hyperkit", "192.168.64.10", 8443)),
 			expected:           true,
 		},
 		{
 			description:        "cannot parse minikube profile list",
 			kubeContext:        "test-cluster",
-			minikubeProfileCmd: testutil.CmdRunOut("minikube profile list -o json", `Random error`),
+			minikubeProfileCmd: testutil.CmdRunOut("minikube profile list -o json --user=skaffold", `Random error`),
 			expected:           false,
 		},
 		{
 			description:        "minikube node ip different from api server url",
 			kubeContext:        "test-cluster",
 			serverURL:          "https://192.168.64.10:8443",
-			minikubeProfileCmd: testutil.CmdRunOut("minikube profile list -o json", fmt.Sprintf(profileStr, "test-cluster", "hyperkit", "192.168.64.11", 8443)),
+			minikubeProfileCmd: testutil.CmdRunOut("minikube profile list -o json --user=skaffold", fmt.Sprintf(profileStr, "test-cluster", "hyperkit", "192.168.64.11", 8443)),
 			expected:           false,
 		},
 	}
 
 	for _, test := range tests {
 		testutil.Run(t, test.description, func(t *testutil.T) {
 			if test.minikubeNotInPath {
-				t.Override(&minikubeBinaryFunc, func() (string, error) { return "", fmt.Errorf("minikube not in PATH") })
+				ver, _ := semver.Make("1.18.0")
+				t.Override(&minikubeBinaryFunc, func() (string, semver.Version, error) {
+					return "", ver, fmt.Errorf("minikube not in PATH")
+				})
 			} else {
-				t.Override(&minikubeBinaryFunc, func() (string, error) { return "minikube", nil })
+				if test.minikuneWithoutUserFalg {
+					ver, _ := semver.Make("1.17.0")
```

Comment:
Use `MustParse()` or 
```suggestion
					ver := semver.Version{Major: 1, Minor: 17, Patch: 0}
```